API/Controllers/FileController.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.EntityFrameworkCore.ChangeTracking.Internal;
+using Models;
+using RestSharp;
+using Services.Services;
+using System.Collections;
+using System.Collections.Specialized;
+using System.IO;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Security.Principal;
+using File = Models.File;
+
+namespace API.Controllers
+{
+    /// <summary>
+    ///     File controller for files
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FileController : ControllerBase
+    {
+
+        private readonly IFileService fileService;
+        private readonly IMapper mapper;
+        private readonly IProjectService projectService;
+        private readonly IUserService userService;
+        private readonly IFileUploader fileUploader;
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileController"/> class.
+        /// </summary>
+        /// <param name="fileService">The file service.</param>
+        /// <param name="mapper">The mapper.</param>
+        /// <param name="projectService">The Project service</param>
+        /// <param name="userService">The User service</param>
+        /// /// <param name="fileUploader">The file uploader extension</param>
+        public FileController(IFileService fileService, IMapper mapper, IProjectService projectService, IFileUploader fileUploader, IUserService userService)
+        {
+            this.fileService = fileService;
+            this.mapper = mapper;
+            this.projectService = projectService;
+            this.userService = userService;
+            this.fileUploader = fileUploader;
+        }
+
+        /// <summary>
+        /// Get all files
+        /// </summary>
+        /// <returns>A response and list of files.</returns>
+        [HttpGet]
+        public async Task<IActionResult> GetFilesAsync()
+        {
+            IEnumerable<File> files = await fileService.GetFilesAsync();
+            if(!files.Any())
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting files.",
+                    Detail = "The database does not contain any files.",
+                    Instance = "47525791-57C4-4DE2-91B1-90086D893112"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(mapper.Map<IEnumerable<File>, IEnumerable<FileResourceResult>>(files));
+        }
+
+        /// <summary>
+        /// Uploads a single file
+        /// </summary>
+        /// <returns>HTTP Response</returns>
+        [HttpPost]
+        [Authorize]
+        public async Task<IActionResult> UploadSingleFile()
+        {
+            IFormFile fileItem = HttpContext.Request.Form.Files["File"];
```

Comment:
I wouldn't straight take it from the request and use modelbinding, I would make a model and let asp.net check it
eg: 
`public async Task<IActionResult> UploadSingleFile([FromForm]File file)`
https://stackoverflow.com/questions/54424712/fromform-with-iformfile-using-asp-net-core

Reply:
Changed it